docs/development/pigs/pig-004.rst

```diff
@@ -0,0 +1,131 @@
+is 157 MB. In ``gammapy-extra/notebooks``, we have ~ 10 tutorial notebooks, each
+20 kB to 1 MB in size, i.e. a few MB in total. Most of the size comes from PNG
+output images in the notebooks, and they usually change on re-run, i.e. even
+though git compresses a bit, the repo grows by up to 1 MB every time a notebook
+is edited. The data files we access from the tutorials are maybe 20 or 30 MB, a
+lot of the datasets we have there are old and should be cleaned up / removed.
+The reason the notebooks and data files were split out was to keep the code
+repository small and avoid it growing to 100 MB or even 1 GB over time.
+
+This separation of code and notebooks and datasets has been a problem in Gammapy
+for years.
+
+Given that Gammapy code changes (and probably always will, even if the pace will
+become slower and slower over the years), the tutorials have to be
+version-coupled to the code. A related question is how tutorial notebooks and
+example datasets are shipped to users. Possible solutions to this and related
+problems have been discussed serval times, see e.g. `GH 1237`_, `GH 1369`_, `GH
+700`_, `GH 431`_, `GH 405`_, `GH 228`_ (probably missed a few).
+
+
+Proposal
+========
+
+This proposal is limited. It outlines a few key changes in the setup for
+notebooks and example data that mostly solve the versioning and shipping issue.
+It does not outline e.g. complete lists of notebooks and datasets, those will be
+iteratively improved in the coming month, with or without an extra PIG.
+
+To solve the versioning issue for notebooks, we propose to move the notebooks
+from ``gammapy-extra/notebooks`` to ``gammapy/notebooks``. Whether to store
+notebooks with or without output cells filled in the repository is a difficult
+decision. Removing output cells before committing has the advantage that the
+files are small, and that the diff in the pull request is also small and review
+becomes possible. On the other hand, the output is not directly visible on
+Github. Note that in any case, a rendered version of the notebooks will be
+available via the docs, that is already in place. We propose to remove the
+output and try that workflow for a while, see if we like it. Since notebooks
+aren't very large, and updates are not needed very often, we think changing to
+notebooks with output would also be acceptable in the ``gammapy`` repository.
+
+To solve the versioning issue for data files, we propose to only use stable
+example datasets. Examples are `gammapy-fermi-lat-data`_ or
+`gammapy-extra/datasets/cta-1dc`_ or the upcoming `HESS DL3 DR1`_ or
+``joint-crab`` datasets. Datasets can be in ``gammapy-extra`` or at any other
+URL, but even if they are in ``gammapy-extra``, they should not be ''live''
+datasets. If an issue is found or something is improved, a separate new dataset
+should be added, instead of changing the existing one. So versioning of example
+datasets is not needed.
+
```

Comment:
This means that only datasets at DL3 can be stored. Reduced datasets such as spectra and reduced IRFs have to be versioned, since they were produced by a version of gammapy.
This raises some questions  about how to organize notebooks and tests.

Reply:
To be more explicit, we should get rid of $GAMMAPY_EXTRA/datasets/hess-crab4_pha/  for instance.
And tests should first create spectral files that will be used later by others tests.

Reply:
There could be higher-level datasets that are stable, not just DL3.
E.g. with `joint-crab` we will have published and archived OGIP spectra.
From Fermi-LAT or HGPS or gamma-cat for a given version we have catalogs.
These files can be used from tutorials.